dracut.sh

```diff
@@ -2166,6 +2176,14 @@ if [[ $kernel_only != yes ]]; then
         fi
     fi
 
+    if [[ $create_sysusers == yes ]] && type -P systemd-sysusers > /dev/null 2>&1; then
```

Comment:
@aafeijoo-suse 

Just an idea: Can you check if "systemd-sysusers" dracut module is NOT included in initrd (but still present in host) instead of a new command line option. I still think that introducing a new command line can be avoided.

Reply:
Sorry, I don't understand. `systemd-sysusers --root="$initdir"` will only create sysusers in the initrd if some module puts sysusers conf in the initrd, but without a new command line option the old method of creating sysusers at every boot would be dropped.

Reply:
> without a new command line option the old method of creating sysusers at every boot would be dropped.

Just trying to understand so that I can review. I do not see this PR changing the 01systemd-sysusers module files. Does that mean that systemd-sysusers (if included in the generated initramfs) will still copy systemd-sysusers binary into initramfs but systemd-sysusers will never be called at boot ?

Reply:
> Just trying to understand so that I can review. I do not see this PR changing the 01systemd-sysusers module files. Does that mean that systemd-sysusers binary (_**do you mean module?**_) (if included in the generated initramfs) will still copy systemd-sysusers binary into initramfs but systemd-sysusers will never be called at boot ?

There is no need to change the systemd-sysusers module if we don't use the kernel command line approach.
```bash
    elif [[ $create_sysusers == yes && $_d_mod == "systemd-sysusers" ]]; then
        continue
```
If the systemd-sysusers module is included in the list of modules to be loaded (`$mods_to_load`) and the `$create_sysusers` switch is set to `yes` (so, create sysusers at generation time), continue to process the next module, so the `install()` method of the systemd-sysusers module is skipped. On the other hand, if the `$create_sysusers` switch is set to `no`, the `install()` method of the systemd-sysusers module will be executed.

Reply:
Thanks for the explanation, it is helpful. I still feel that a new dracut command line option for this feature is a significant maitenance burdon for very little benefit. I do not think maintaining the "old way" is necessary.

A better solution would be to somehow implement this feature in the systemd-sysusers dracut module and not in dracut itself. 

It is a confusing concept for me (and maybe also for users) that a module is included but never actually installed based on some dracut command line option.

Reply:
Thanks for you review. That's why this is an RFC, we need to find the best approach to implement this.

Another option to keep the systemd-sysusers (at least listed as installed) is moving the part of my previous comment to it. Something like:

```diff
diff --git a/modules.d/01systemd-sysusers/module-setup.sh b/modules.d/01systemd-sysusers/module-setup.sh
index 4ec48dc6..488be0e9 100755
--- a/modules.d/01systemd-sysusers/module-setup.sh
+++ b/modules.d/01systemd-sysusers/module-setup.sh
@@ -24,6 +24,10 @@ depends() {
 # Install the required file(s) and directories for the module in the initramfs.
 install() {
 
+    # Do not install systemd-sysusers stuff in the initramfs if dracut creates
+    # the system users only when it's generated.
+    [[ $create_sysusers == yes ]] && return 0
+
     inst_simple "$moddir/sysusers-dracut.conf" "$systemdsystemunitdir/systemd-sysusers.service.d/sysusers-dracut.conf"
 
     inst_multiple -o \
```

This would keep both ways of working. To remove the old method, we just have to remove the `install()` method from systemd-sysusers.

I'm not sure whether we should not keep the "old way", maybe the systemd-sysusers infrastructure might be needed in the initrd in some use case.

Reply:
We can always reintroduce the "old way" if we understand better why it is needed, but why to maintain it just speculatively ? 

One option (not great) to maintain the "old way" is to have 2 modules - one for the "old way" and one for the "new way".

Reply:
One more thoughts on this: The "old way" is really just installing some files into initramfs, which can be already archived by existing command line options (-I and -i) without any additional explicit dracut module. Perhaps this makes you more comfortable not maintaining the "old" way any longer.